### ISSUE TITLE
Introduce a concept of a Soft Deprecation (or an Informative Note) encouraging, but not mandating a change

### ISSUE THREAD

**Issue by anna-geller** [kestra-io/kestra#4753]
For changes with aliases such as renamed plugins or properties, we may keep the old names via aliases without ever removing them. Therefore, treating those as true deprecations can unnecessarily worry our users. 

We should treat those differently in the UI:
1. `@Deprecated(forRemoval = true)` - urgent changes that require attention — we should display them as warnings in the UI (as we currently do)
2. `@Deprecated` - non-urgent changes that user should be aware of — we should display those with an info icon, informing the user about such "Soft Deprecation" 

In contrast to a true deprecation warning, a soft deprecation indicates *a gentle nudge* encouraging but not mandating a change.

## Context: Why the change

1. We've got support tickets and Slack help requests from users worrying they need to rewrite their flows to match some new naming even though they can take as long as they need - with aliases, there's no rush to update code using the old names
3. With too many warnings for simple issues such as renaming, users may start ignoring warnings, which would be disastrous.

## TL;DR

- [ ] adjust the annotation for urgent (marked for removal) vs non-urgent changes (e.g. renamed plugins)
- [ ] new UI handling of such soft deprecations with an info icon

**Comment by tchiotludo** [comment]
Perfect on the ui side, for the code side I think the best is : 
- `@Deprecated(forRemoval = true)`: urgent changes
- `@Deprecated`: non-urgent changes

This one is standard in java

**Comment by loicmathieu** [comment]
Aliases are now moved to an information notice, they were the most source of warnings.

I checked the usage of deprecation, they are almost all for Plugin sor properties that we want to delete some day, but none are using `@Deprecated(forRemoval = true)` so if we want to handle differently `@Dreprecated` and `@Deprecated(forRemoval = true)` we would also need to review all our usage of `@Deprecated` on all plugins.

So the effort to implement soft deprecation for `@Deprecated(forRemoval = true)` may not worth it at this stage.

**Comment by Ben8t** [comment]
Looks ok to close. @anna-geller wdyt ?

**Comment by anna-geller** [comment]
> we would also need to review all our usage of @Deprecated on all plugins.

We should do that

`@Deprecated` is a soft deprecation we handle with information icon ℹ️

`@Deprecated(forRemoval=true)` is a real deprecation which emits warning ⚠️ 

We should validate all those and check if the soft/hard deprecation makes sense

Likely more product work than engineering to compile a table and review together - @Ben8t would you be open to tackle it? Definitely not urgent, can be done in 0.21 or later 

Agree the UI part is done 👍

**Comment by Ben8t** [comment]
Yes sounds great ! Added to my watch 👍 (will try to do this during QA prior to 0.20 release)

**Comment by anna-geller** [comment]
ok seems I was wrong

![image](https://github.com/user-attachments/assets/afa4bcb5-5fde-44c4-a858-0b2091698649)
so far, Loic seems right that none of them are using the deprecation for removal so **everything is a soft deprecation for now** and the UI now treats those as information, not a warning, so we are good 

We probably need to just clarify in the team to **use real deprecation more often** for things that we want to remove one day
